distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1543,6 +1519,252 @@ public void testFederatedDownloadProductToCacheOnlyCacheDisabled() throws Except
+                filename,
+                String.format("Resource retrieval completed, %d bytes retrieved. ",
+                        resourceData.length()),
+                "COMPLETE");
+
+        List<String> localhostActivities = localhostCometDClient.getMessagesInAscOrder(
+                ACTIVITIES_CHANNEL);
+        assertThat(localhostActivities.size(), is(2));
+        CometDMessageValidator.verifyActivity(JsonPath.from(localhostActivities.get(0)),
+                filename2,
+                "Resource retrieval started. ",
+                "STARTED");
+        CometDMessageValidator.verifyActivity(JsonPath.from(localhostActivities.get(1)),
+                filename2,
+                String.format("Resource retrieval completed, %d bytes retrieved. ",
+                        resourceData2.length()),
+                "COMPLETE");
+    }
+
+    @Test
+    public void testSingleUserDownloadSameProductSyncAndAsync() throws Exception {
+        getSecurityPolicy().configureWebContextPolicy(null,
+                "/=SAML|basic,/solr=SAML|PKI|basic",
+                null,
+                null);
+
+        adminCometDClient = setupCometDClientWithUser(Arrays.asList(NOTIFICATIONS_CHANNEL,
+                ACTIVITIES_CHANNEL), "admin", "admin");
+
+        String filename = "product4.txt";
+        String metacardId = generateUniqueMetacardId();
+        String resourceData = getResourceData(metacardId);
+
+        setupStubCswResponse(filename, metacardId, resourceData);
+
+        String resourceDownloadUrlLocalhostUserSync =
+                REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + metacardId
+                        + "?transform=resource";
+
+        String resourceDownloadUrlLocalhostUserAsync =
+                RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl() + "?source=" + CSW_STUB_SOURCE_ID
+                        + "&metacard=" + metacardId;
+
+        // Download product via async and then sync, should only call the stub server to download once
+        // @formatter:off
+        String downloadId = given().auth().preemptive().basic("admin", "admin").when()
+                .get(resourceDownloadUrlLocalhostUserAsync).then().log().all()
+                .extract().jsonPath().getString("downloadId");
+        assertThat(downloadId, is(notNullValue()));
+        given().auth().preemptive().basic("admin", "admin").when()
+                .get(resourceDownloadUrlLocalhostUserSync).then().log().all()
+                .assertThat().contentType("text/plain").body(is(resourceData));
+
+        verifyCswStubCall(1, metacardId);
+
+        expect("Waiting for notifications. Received " + adminCometDClient.getMessages(
+                NOTIFICATIONS_CHANNEL)
+                .size() + " of 1").within(10, SECONDS)
+                .until(() -> adminCometDClient.getMessages(NOTIFICATIONS_CHANNEL)
+                        .size() == 1);
+        expect("Waiting for activities. Received " + adminCometDClient.getMessages(
```

Comment:
We could consider takingout the expect activities since we don't assert against any of the messages.